app/assets/scripts/utils/map-utils.js

```diff
@@ -18,20 +21,44 @@ module.exports.getProjectCentroids = function (projects, features) {
     let locations = get(project, 'location');
     if (locations && Array.isArray(locations)) {
       locations.forEach(function (location) {
-        // TODO look at district or marker to see if there's more granular data
-        const region = location.district.governorate;
-        regions[region] = regions[region] || [];
-        regions[region].push(project);
+        const region = location.district;
+        let type;
+        region.district.length && region.district.toLowerCase() !== 'all'
+          ? type = 'district'
+          : type = 'governorate';
+        const regionId = region[type];
+        regions[regionId] = regions[regionId] || {type: type, regions: []};
+        regions[regionId].regions.push(project);
       });
     }
   });
 
-  Object.keys(regions).forEach(function (id) {
-    const meta = governorates.byId(id);
-    const feature = features.find((f) => f.properties.admin_id === meta.egy);
+  let {districts, egy2} = features;
+  districts = districts.features;
+  egy2 = egy2.features;
+  console.log(districts.map((d) => d.properties.Qism_Mar_1))
+  Object.keys(regions).forEach(function (id, i) {
+    let meta;
+    let feature;
+    const type = regions[id].type;
+    // console.log(type)
+    if (type === 'district') {
```

Comment:
The script fails on the first district, ID `1501`. I have some print statements around the operation to show the district ID going in, and I also hardcoded a test of `1501` into the districts util, which also fails.